app/src/main/java/com/codesoom/assignment/services/TaskService.java

```diff
@@ -0,0 +1,59 @@
+package com.codesoom.assignment.services;
+
+import com.codesoom.assignment.utils.IdGenerator;
+import com.codesoom.assignment.exceptions.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import com.codesoom.assignment.repositories.TaskRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TaskService {
+
+    private final TaskRepository taskRepository;
+    private final IdGenerator idGenerator;
+
+    public TaskService(TaskRepository taskRepository, IdGenerator idGenerator) {
+        this.taskRepository = taskRepository;
+        this.idGenerator = idGenerator;
+    }
+
+    public List<Task> getTasks() {
+        return taskRepository.findAll();
+    }
+
+    public Task addTask(Task task) {
+        Long newId = idGenerator.generateId();
+        task.setId(newId);
+
+        return taskRepository.save(task);
+    }
+
+    public Task getTask(Long id) {
+        Optional<Task> task = taskRepository.findOne(id);
+
+        return task.orElseThrow(()-> new TaskNotFoundException("Not found task with id = " + id));
+    }
+
+    public Task updateTask(Long id, Task newTask) {
+        Task task = getTask(id);
+        task.update(newTask);
+
+        return taskRepository.save(task);
+    }
+
+    public void deleteTask(Long id) {
+        if (!isExist(id)) {
+            throw new TaskNotFoundException("Not found task with id = " + id);
```

Comment:
일반적인 예외 메시지는 크게 두 종류로 나뉩니다.

1. ~이면 안됩니다 / ~은 불가능합니다.
2. ~을 할 수 없습니다. / ~이 없습니다.
3. ~하기 때문에 ~을 할 수 없습니다.

위에서 작성하신 에러 메시지는 2번에 해당합니다. 그런데 개념상의 정의가 아닌 사용자의 특정 행동에 예외 처리를 할 때에는 3번과 같이 작성해주는 경우가 도움될 수도 있습니다.

다음은 제가 작성한 에러 메시지입니다. 두 방식을 비교해 보고 고민해보시고, 다른 예외는 어떻게 메시지를 수정할 것인지에 대해서도 생각해 보세요.

```suggestion
            throw new TaskNotFoundException("존재하지 않는 task id가 주어졌으므로 task를 삭제할 수 없습니다. 문제의 id = " + id);
```